### ISSUE TITLE
Teuchos: class std::auto_ptr' is deprecated

### ISSUE THREAD

**Issue by nschloe** [trilinos/Trilinos#13]
Since C++11, [auto_ptr is deprecated](http://www.cplusplus.com/reference/memory/auto_ptr/); Teuchos still uses it:

```
/«PKGBUILDDIR»/packages/teuchos/comm/src/Teuchos_CommHelpers.cpp:232:12: warning: 'template<class> class std::auto_ptr' is deprecated [-Wdeprecated-declarations]
```

**Comment by bartlettroscoe** [comment]
We are not allowed to use C++11 in Teuchos yet.  We would have to ifdef this to use std::unique_ptr when C++11 is enabled.

**Comment by rppawlo** [comment]
To follow up on Ross's comment, while Trilinos has officially switched to requiring c++11, we have some customers using the old stack epetra, aztecoo, teuchos, nox, ... that could not upgrade compilers on all of their hardware fast enough.  So we have agreed to hold a few packages back to compile to the old standard for another year so until that they can sort things out.

**Comment by nschloe** [comment]
Has the status of this changed yet?

**Comment by mhoemmen** [comment]
> Has the status of this changed yet?

No.  Would welcome a patch as long as it has both a C++11 and a C++98 code path (via ifdefs).

**Comment by aprokop** [comment]
Fixed in #1319.
